### ISSUE TITLE
Doesn't seem to read tsconfig value from loader string?

### ISSUE THREAD

**Issue by supermoos** [s-panferov/awesome-typescript-loader#59]
I have this setup:
    module: {
        loaders: [
            {test: /\.ts$/, loader: 'awesome-typescript-loader?tsconfig=tsconfigwebpack.json', exclude: /node_modules/}
]
}

But it still uses the tsconfig.json instead of the one I have defined for compilation?

**Comment by WebWalks** [comment]
I have also found this problem:
The problem is due to an issue in dist/index.js (in the npm node_module folder).
line 122: (line 191 in the index.ts)
    var configFileName = tsImpl.findConfigFile(options.tsconfig || process.cwd());

The actual findConfigFile method of Typescript (in 1.6.2 for sure) doesn't actually do anything with the passed folder, apart from reduce it folder by folder (it just works down to the root from the current folder looking for a tsconfig.json)

My temp solution is to pass the full path to the tsconfig as the loader param and replace the offending line with:
    var configFileName = options.tsconfig;

**Comment by supermoos** [comment]
Arh okay. Could you post your loader config for webpack?

**Comment by WebWalks** [comment]
Hope this helps

var path = require("path");
var tsConfigPath = path.join(__dirname, "source", "tsconfig.json");
var entryFile = path.join(__dirname, "source", "browser.tsx");
var outputPath = path.join(__dirname, "public", "js");

module.exports = {
  resolve: {
    extensions: ["", ".ts", ".tsx", ".webpack.js", ".web.js", ".js"]
  },
  entry: [
    entryFile
  ],
  output: {
    path: outputPath,
    filename: "bundle.js",
    publicPath: "/js/"
  },
  module: {
    loaders: [{
      test: /\.ts(x)?$/,
      loader: "awesome-typescript-loader?tsconfig=" + tsConfigPath + "&emitRequireType=false"
    }]
  }
};

**Comment by santiagoaguiar** [comment]
Had the same issue. 

Was very confusing because the path up to the directory is honored but the filename isn't.

I changed:
```typescript
    let folder = options.tsconfig || process.cwd();
    configFilePath = tsconfig.resolveSync(folder);
    if (configFilePath) {
        configFile = parseContent(fs.readFileSync(configFilePath).toString(), configFilePath);
```
To:
```typescript
    let configFilePath = options.tsconfig || tsconfig.resolveSync(process.cwd());
    if (configFilePath) {
        configFile = parseContent(fs.readFileSync(configFilePath).toString(), configFilePath);
```

Which works for me, but might break some existing code; however I'm not sure how it can be useful by only letting you pick a different folder.

I'm currently having a `tsconfig.json` & a `tsconfig.webpack.json` on my `src/` folder and wan't to use `tsconfig.webpack.json` from the loader.

**Comment by Cleod9** [comment]
I can confirm this issue, @santiagoaguiar your fix works.

Any chance you can submit a PR with that change? I see no reason not to merge it in. I doubt many people are using the feature in the first place judging by the amount of attention this thread has gotten.

**Comment by s-panferov** [comment]
Hi all, thanks for your report. I've never thought about this use-case with different config files for different build systems. Could you please explain why do you need two configs?

Anyway, I'll fix this ASAP.

**Comment by Cleod9** [comment]
Thanks! What I wanted was to have separate configs so I could enable/disable `babel` as a pass through. I use the `config` module for Node.js to drive my build settings, and it let's me specify configurations based on the environment

**Comment by s-panferov** [comment]
Landed in v0.16.0
